test/test.compute.ts

```diff
@@ -66,6 +66,40 @@ describe('Compute auth client', () => {
     });
   });
 
+  it('should refresh if access token will expired soon and time to refresh' +
+         ' before expiration is set',
+     (done) => {
+       nock('http://metadata.google.internal')
+           .get(
+               '/computeMetadata/v1beta1/instance/service-accounts/default/token')
+           .reply(200, {access_token: 'abc123', expires_in: 10000});
+       compute = new Compute({refreshTokenEarlyMillis: 10000});
+       compute.credentials.access_token = 'initial-access-token';
+       compute.credentials.expiry_date = (new Date()).getTime() + 5000;
+       compute.request({url: 'http://foo'}, () => {
+         assert.equal(compute.credentials.access_token, 'abc123');
+         done();
+       });
+     });
+
+  it('should not refresh if access token will not expire soon and time to' +
+         ' refresh before expiration is set',
+     (done) => {
+       const scope =
+           nock('http://metadata.google.internal')
+               .get(
+                   '/computeMetadata/v1beta1/instance/service-accounts/default/token')
+               .reply(200, {access_token: 'abc123', expires_in: 10000});
+       compute.credentials.access_token = 'initial-access-token';
+       compute.credentials.expiry_date = (new Date()).getTime() + 11000;
+       compute.request({url: 'http://foo'}, () => {
```

Comment:
async/await

Reply:
I believe that this is an easy change in test.compute.ts or in test.jwt.ts.

When I re-wrote this test as 
```
  it('should not refresh if access token will not expire soon and time to' +
         ' refresh before expiration is set',
     async () => {
       const scope =
           nock('http://metadata.google.internal')
               .get(
                   '/computeMetadata/v1beta1/instance/service-accounts/default/token')
               .reply(200, {access_token: 'abc123', expires_in: 10000});
       compute.credentials.access_token = 'initial-access-token';
       compute.credentials.expiry_date = (new Date()).getTime() + 11000;
       await compute.request({url: 'http://foo'});
       assert.equal(compute.credentials.access_token, 'initial-access-token');
       nock.cleanAll();
     });
```

it failed with:
```
NetConnectNotAllowedError: Nock: Not allow net connect for "foo:80/"
      at node_modules/nock/lib/intercept.js:404:21
      at Object.module.request (node_modules/nock/lib/common.js:140:14)
      at RedirectableRequest._performRequest (node_modules/follow-redirects/index.js:73:20)
      at new RedirectableRequest (node_modules/follow-redirects/index.js:57:7)
      at Object.wrappedProtocol.request (node_modules/follow-redirects/index.js:229:10)
      at dispatchHttpRequest (node_modules/axios/lib/adapters/http.js:132:25)
      at new Promise (<anonymous>)
      at httpAdapter (node_modules/axios/lib/adapters/http.js:18:10)
      at dispatchRequest (node_modules/axios/lib/core/dispatchRequest.js:59:10)
      at <anonymous>
```

I have changed tests in test.oauth2.ts